cunumeric/logic.py

```diff
@@ -0,0 +1,281 @@
+    --------
+    Multiple GPUs, Multiple CPUs
+
+    """
+    x = convert_to_cunumeric_ndarray(x)
+    if out is not None:
+        out = convert_to_cunumeric_ndarray(out, share=True)
+    rhs1 = isinf(x)
+    rhs2 = signbit(x)
+    return logical_and(rhs1, rhs2, out=out)
+
+
+def isposinf(x, out=None):
+    """
+
+    Test element-wise for positive infinity, return result as bool array.
+
+    Parameters
+    ----------
+    x : array_like
+        The input array.
+    out : array_like, optional
+        A location into which the result is stored. If provided, it must have a
+        shape that the input broadcasts to. If not provided or None, a
+        freshly-allocated boolean array is returned.
+
+    Returns
+    -------
+    out : ndarray
+        A boolean array with the same dimensions as the input.
+        If second argument is not supplied then a boolean array is returned
+        with values True where the corresponding element of the input is
+        positive infinity and values False where the element of the input is
+        not positive infinity.
+
+        If a second argument is supplied the result is stored there. If the
+        type of that array is a numeric type the result is represented as zeros
+        and ones, if the type is boolean then as False and True.
+        The return value `out` is then a reference to that array.
+
+    See Also
+    --------
+    numpy.isposinf
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+
+    """
+    x = convert_to_cunumeric_ndarray(x)
+    if out is not None:
+        out = convert_to_cunumeric_ndarray(out, share=True)
+    rhs1 = isinf(x)
+    rhs2 = ~signbit(x)
+    return logical_and(rhs1, rhs2, out=out)
+
+
+def iscomplex(x):
+    """
+
+    Returns a bool array, where True if input element is complex.
+
+    What is tested is whether the input has a non-zero imaginary part, not if
+    the input type is complex.
+
+    Parameters
+    ----------
+    x : array_like
+        Input array.
+
+    Returns
+    -------
+    out : ndarray[bool]
+        Output array.
+
+    See Also
+    --------
+    numpy.iscomplex
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+
+    """
+    x = convert_to_cunumeric_ndarray(x)
+    if x.dtype.kind != "c":
+        return full(x.shape, False, dtype=bool)
+    else:
+        return x.imag != 0
+
+
+def iscomplexobj(x):
+    """
+
+    Check for a complex type or an array of complex numbers.
+
+    The type of the input is checked, not the value. Even if the input
+    has an imaginary part equal to zero, `iscomplexobj` evaluates to True.
+
+    Parameters
+    ----------
+    x : any
+        The input can be of any type and shape.
+
+    Returns
+    -------
+    iscomplexobj : bool
+        The return value, True if `x` is of a complex type or has at least
+        one complex element.
+
+    See Also
+    --------
+    numpy.iscomplexobj
+
+    Availability
+    --------
+    Single CPU
```

Comment:
Technically this is available everywhere?

Reply:
I guess you fall back to NumPy on some occasions, so this field is ill-defined?

Reply:
Yes, that field is not well-defined for things that are inherently sequential, but I feel Single CPU is a correct way to characterize them.